recipe/node.go

```diff
@@ -0,0 +1,87 @@
+package recipe
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+// RecipeNode contains the json data for a recipe node
+type RecipeNode struct {
+	Name       yaml.Node    `json:"name" yaml:"name"`
+	Source     PluginNode   `json:"source" yaml:"source"`
+	Sinks      []PluginNode `json:"sinks" yaml:"sinks"`
+	Processors []PluginNode `json:"processors" yaml:"processors"`
+}
+
+// PluginNode contains the json data for a recipe node that is being used for
+// generating the plugins code for a recipe.
+type PluginNode struct {
+	Name   yaml.Node            `json:"name" yaml:"name"`
+	Config map[string]yaml.Node `json:"config" yaml:"config"`
+}
+
+// decodeConfig decodes the plugins config
+func (plug *PluginNode) decodeConfig() map[string]interface{} {
+	config := make(map[string]interface{})
+
+	for key, val := range plug.Config {
+		config[key] = val.Value
+	}
+	return config
+}
+
+// toRecipe passes the value from RecipeNode to Recipe
+func (node RecipeNode) toRecipe() (recipe Recipe, err error) {
```

Comment:
if the operation does not mutate some fields in the struct, we can use reference here instead of value.
```suggestion
func (node *RecipeNode) toRecipe() (recipe Recipe, err error) {
```

Reply:
CMIIW but shouldn't the reason to use pointer is because we want to mutate? 🤔

Reply:
Oh yes that's correct. Should be the other way around. Sorry for mixing this up 😁.
Thanks for pointing out @StewartJingga.
All good here @scortier